packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -90,28 +122,15 @@ export class State {
       eventName: options.eventName ?? `${this.stateName}_to_${targetState.stateName}`,
       nextState: targetState,
       condition: options.when,
+      actions: options.actions,
     });
   }
 
   /**
-   * Collect states in dependency gragh that constructed by state transitions,
-   * and return the JSONs of the states.
-   * This function is called recursively and collect the states.
-   *
-   * @internal
+   * Return the JSONs of the states in dependency gragh that constructed by state transitions
    */
-  public _collectStateJsons(collectedStates: Set<State>): CfnDetectorModel.StateProperty[] {
-    if (collectedStates.has(this)) {
-      return [];
-    }
-    collectedStates.add(this);
-
-    return [
-      this.toStateJson(),
-      ...this.transitionEvents.flatMap(transitionEvent => {
-        return transitionEvent.nextState._collectStateJsons(collectedStates);
-      }),
-    ];
+  public bind(detectorModel: IDetectorModel): CfnDetectorModel.StateProperty[] {
```

Comment:
Can we revert these changes please?

This is "future proofing", since we're not actually using the `detectorModel` parameter anyway yet (and, like I said above, I'm pretty sure it's actually incorrect to pass it here, because it has not been initialized).

That's why I don't like creating APIs that are not used - we are doing weird guesses here, and changing a lot of code, that, in my opinion, will not work anyway.

So, please revert the changes here, and let's stop the guessing game.

Reply:
OK! I'd start by creating `aws-iotevents-actions`. Then I will implement some actions actually without guess.